doc/changelog.md

```diff
@@ -94,6 +95,9 @@
 * Small changes to make pylint==3.2.0 succeed.
   [(#739)](https://github.com/PennyLaneAI/catalyst/pull/739)
 
+* Callback refactoring.
```

Comment:
What is the goal of the refactoring / what did it achieve?

Reply:
I did a rebase and added these changes to the changelog: https://github.com/PennyLaneAI/catalyst/pull/742/commits/0268382b60f0bba094da9b7ea5e4581dd03fd72d

Reply:
Also, I should note that this is no longer strictly necessary. It would be necessary if we had kept with the "fwd and bwd passes are also callbacks" approach. But I really like this clean up. I think it makes the whole code a lot more readable.